src/components/number/number.pw.tsx

```diff
@@ -0,0 +1,685 @@
+  test("should call onChange callback when a type event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    const inputValue = "1";
+    let callbackCount = 0;
+    await mount(
+      <NumberInputComponent
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await commonDataElementInputPreview(page).type(inputValue);
+    expect(callbackCount).toBe(1);
+  });
+
+  ["ArrowDown", "ArrowLeft", "ArrowRight", "ArrowUp"].forEach((key) => {
+    test(`should call onKeyDown callback when ${key} key is triggered`, async ({
+      mount,
+      page,
+    }) => {
+      let callbackCount = 0;
+      await mount(
+        <NumberInputComponent
+          onKeyDown={() => {
+            callbackCount += 1;
+          }}
+        />
+      );
+
+      const elementInput = commonDataElementInputPreview(page);
+      await elementInput.focus();
+      await elementInput.press(key);
+
+      expect(callbackCount).toEqual(1);
+    });
+  });
+
+  test("should call onBlur callback when a blur event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <NumberInputComponent
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const elementInput = commonDataElementInputPreview(page);
+    await elementInput.click();
+    await elementInput.blur();
+
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onMouseDown callback when a mousedown event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <NumberInputComponent
+        onMouseDown={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const elementInput = commonDataElementInputPreview(page);
+    await elementInput.dispatchEvent("mousedown");
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("check accessibility", () => {
+  test("should pass accessibility tests for default component", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Default />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for sizes example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Sizes />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for disabled example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Disabled />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for read only example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ReadOnly />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for autofocus example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<AutoFocus />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for with label inline example", async ({
```

Comment:
```suggestion
  test("should pass accessibility tests for label inline example", async ({
```

**suggestion**. Same in the next 6 tests.

Reply:
turns out this is OK as it's due to the name of the story being "with label inline" etc